Simplenote/Classes/InterlinkViewController.swift

```diff
@@ -0,0 +1,184 @@
+        backgroundLayer.shadowOpacity = Shadow.opacity
+        backgroundLayer.shadowOffset = Shadow.offset
+    }
+
+    func setupTableView() {
+        tableView.register(Value1TableViewCell.self, forCellReuseIdentifier: Value1TableViewCell.reuseIdentifier)
+        tableView.backgroundColor = .clear
+        tableView.separatorColor = .simplenoteDividerColor
+        tableView.tableFooterView = UIView()
+        tableView.layer.cornerRadius = Metrics.cornerRadius
+    }
+}
+
+
+// MARK: - UITableViewDataSource
+//
+extension InterlinkViewController: UITableViewDataSource {
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        notes.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let note = notes[indexPath.row]
+        note.ensurePreviewStringsAreAvailable()
+
+        let tableViewCell = tableView.dequeueReusableCell(ofType: Value1TableViewCell.self, for: indexPath)
+        tableViewCell.title = note.titlePreview
+        tableViewCell.backgroundColor = .clear
+        tableViewCell.separatorInset = .zero
+
+        return tableViewCell
+    }
+}
+
+
+// MARK: - Geometry
+//
+private extension InterlinkViewController {
+
+    /// Updates the layout constraints so that the receiver shows up **around** the specified Keyword in the specified TextView
+    ///
+    func refreshConstraints(keywordRange: Range<String.Index>, in textView: UITextView) {
+        let targetHeight = calculateHeight()
+        let targetLocation = calculateLocation(for: targetHeight, around: keywordRange, in: textView)
+        let targetPadding = textView.textContainer.lineFragmentPadding
+
+        tableTopConstraint.constant = targetLocation
+        tableHeightConstraint.constant = targetHeight
+        tableLeadingConstraint.constant = targetPadding
+        tableTrailingConstraint.constant = targetPadding
+    }
+
+    /// Starts tracking ContentOffset changes in our sibling TextView
+    ///
+    func startObservingContentOffset(in textView: UITextView) {
+        kvoOffsetToken = textView.observe(\UITextView.contentOffset, options: [.old, .new]) { [weak self] (_, value) in
+            guard let oldY = value.oldValue?.y, let newY = value.newValue?.y, oldY != newY else {
+                return
+            }
+
+            self?.tableTopConstraint.constant += oldY - newY
+        }
+    }
+
+    /// Returns the target Origin.Y
+    /// -   Parameters:
+    ///     - height: The new target height
+    ///     - range: Anchor's Range
+    ///     - textView: Sibling TextView
+    ///
+    /// - Important: We'll always prefer either the "location above the cursor" whenever the **Maxed Out TableView** can be fit (that is: +3.5 results!)
+    ///
+    func calculateLocation(for height: CGFloat, around range: Range<String.Index>, in textView: UITextView) -> CGFloat {
+        let anchorFrame = textView.locationInSuperviewForText(in: range)
```

Comment:
Do we need to convert it to a local view coordinates?

Reply:
The TextView itself would be a sibling of this ViewController's view (both live in the same superview).

Since this VC is expected to assume the exact same size as it's parent, there should be no need for conversion (AFAIK!!), please correct me if I'm wrong!

Reply:
I would like to make as little assumptions as possible. I think safest approach would be if `textView.locationInSuperviewForText(in: range)` returns frame in Window coordinates and we would convert it back to our local coordinates here (similar to how we do it for keyboard frame).